### PR TITLE
fix unweighted flag saving

### LINKED ISSUE #58: `save_weighted_flag` crashes (and is uncovered by tests)
#### Description
`save_weighted_flag` throws error instead of doing it's job.

#### Steps/Code to Reproduce
1) Have a binary adjacency matrix `G`
2) call `save_weighted_flag('foo.bar', G)`

#### Expected Results
saves flag to `foo.bar`

#### Actual Results
crashes:
```
Traceback (most recent call last):
  File "/home/florian/project/digraph-analyzer/playground.py", line 14, in <module>
    save_unweighted_flag('/tmp/bbp_1.flag', G)
  File "/home/florian/.local/lib/python3.9/site-packages/pyflagser/flagio.py", line 196, in save_unweighted_flag
    np.savetxt(f, edges, comments='', header='dim 1', fmt='%i %i')
  File "<__array_function__ internals>", line 5, in savetxt
  File "/usr/lib/python3.9/site-packages/numpy/lib/npyio.py", line 1404, in savetxt
    raise error
ValueError: fmt has wrong number of % formats:  %i %i
```

#### Versions
everything is fresh from the git

#### Fix:
The following diff fixes this:
```
[florian@frankenstein-archlinux pyflagser]$ git diff
diff --git a/pyflagser/flagio.py b/pyflagser/flagio.py
index 7f3ebc9..446bf17 100644
--- a/pyflagser/flagio.py
+++ b/pyflagser/flagio.py
@@ -191,9 +191,8 @@ def save_unweighted_flag(fname, adjacency_matrix):
     vertices, edges = _extract_unweighted_graph(adjacency_matrix)
 
     with open(fname, 'w') as f:
-        np.savetxt(f, vertices, delimiter=' ', comments='', header='dim 0',
-                   fmt='%.18e')
-        np.savetxt(f, edges, comments='', header='dim 1', fmt='%i %i')
+        np.savetxt(f, vertices.reshape((1,-1)), delimiter=' ', header='dim 0', fmt='%i')
+        np.savetxt(f, edges, comments='', header='dim 1', fmt='%i %i %i')
 
 
 def save_weighted_flag(fname, adjacency_matrix, max_edge_weight=None):
```

I'll open a PR soon. First need to figure out how :D

### PR DESCRIPTION
**Pull request by flomlo**
#### Reference Issues/PRs
Example: Fixes #58 partially, does not add test coverage yet

#### What does this implement/fix? Explain your changes.
now looks more like `save_unweighted_flag`. also works.

### CODE REVIEW

**Comment by ulupo** [comment]
This is a portion of the code I am not very familiar with unfortunately.  @flomlo could you perhaps explain the changes, why the previous version failed, etc?  Also it would be good to add a little regression tests as discussed in #58.  It can be quite simple.

**Comment by flomlo** [comment]
Hi! Sorry, haven't noticed before that the discussion moved here.

@ already submitted code:
I've modified the `save_unweighted_flag` routine to make the output of it look more like e.g. `e.flag` in the flagser` repo: https://github.com/luetge/flagser/blob/7ffe92782ae639b4265a52403d2d90d3151bb67b/test/e.flag

There are two changes:
  - `vertices` is reshaped so that it forms a line vector instead of a column vector. This is analogous to save_weighted_flag`. If one does not do this, the outputted flag file is corrupted and it complains about not enough vertices being available for the respective edges.

  - `edges` are saved with `fmt='%i %i %i'` instead of `fmt='%i %i`. Here the first integer is `i`, the second `j` and the third the weight `w_ji`. Well, the original version simply did not work, as `edges` has three columns. 

Note that the resulting .flag files have 1 as weights, not 0. That does not matter, though.

**Comment by flomlo** [comment]
Regarding the test cases:

For some reason only `save_weighted_flag` is tested. I think just copying that to `save_unweighted_flag` is good enough. I do not really understand why the current testing code is written like it is written, so I would prefer to not modify it.

**Comment by ulupo** [comment]
@flomlo thanks for submitting this!  Let me add the test then, and I'll merge.

**Comment by ulupo** [comment]
I notice that you made your changes in your `master` branch and created a PR from there.  I know it seems overly pedantic to say this, but this is strongly recommended *against*.  A separate "feature branch" should be created with the changes and the PR should come from there. `master` in a fork should always be in sync with the origin repository.  Feel free to either close this PR and reopen a new one from such a feature branch, or to try one of the solutions in https://coding.abel.nu/2015/03/fixing-a-pull-request-from-master/ (probably not worth the trouble in this case).  Sorry for being annoying!

**Comment by flomlo** [comment]
regarding tests: 

I would have written something which checks that (for a random graph `g`) the following holds:
load(save(g)) = g.
(This should be true both for weighted and unweighted.)

The testsuite does other stuff, and I'm not sure why

**Comment by flomlo** [comment]
> @flomlo I notice that you made your changes in your `master` branch and created a PR from there. I know it seems overly pedantic to say this, but this is strongly recommended _against_. A separate "feature branch" should be created with the changes and the PR should come from there. `master` in a fork should always be in sync with the origin repository. Feel free to either close this PR and reopen a new one from such a feature branch, or to try one of the solutions in https://coding.abel.nu/2015/03/fixing-a-pull-request-from-master/ (probably not worth the trouble in this case). Sorry for being annoying!


Yeah, I'm git-noob :D 
I'll check on it after lunch break, ok?

**Comment by ulupo** [comment]
@flomlo no worries!

**Comment by flomlo** [comment]
Oh, and another quick note regarding the tests: I'm not sure about the expected behaviour of saving/loading of graphs with zero edges. This might be an edge-case to check for (or zero-edge-case?)
